### PR TITLE
feat: Add alert dialog whenever register button is clicked and 0 tickets are selected

### LINKED ISSUE #359: Show alert dialog whenever user selects 0 tickets and clicks on register
Working on it :+1:

### PR DESCRIPTION
**Pull request by dreadpool2**
Fixes #359

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 1]
```diff
@@ -89,12 +90,16 @@ class TicketsFragment : Fragment() {
...
-            bundle.putSerializable(TICKET_ID_AND_QTY, tickeIdAndQty)
-            fragment.arguments = bundle
-            activity?.supportFragmentManager?.beginTransaction()?.replace(R.id.rootLayout, fragment)?.addToBackStack(null)?.commit()
+            if (totalTickets() != 0) {
+                val fragment = AttendeeFragment()
+                val bundle = Bundle()
+                bundle.putLong(EVENT_ID, id)
+                bundle.putSerializable(TICKET_ID_AND_QTY, tickeIdAndQty)
+                fragment.arguments = bundle
+                activity?.supportFragmentManager?.beginTransaction()?.replace(R.id.rootLayout, fragment)?.addToBackStack(null)?.commit()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: Multiline

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 2]
```diff
@@ -133,4 +138,16 @@ class TicketsFragment : Fragment() {
         rootView.time.text = EventUtils.getFormattedDateTimeRangeDetailed(startsAt, endsAt)
     }
 
+    private fun handleNoTicketsSelected() {
+        val builder = AlertDialog.Builder(activity)
+        builder.setMessage(activity?.resources?.getString(R.string.no_tickets_message))
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: No need to get activity to get string

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: Will do the changes :+1:

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: PR Updated

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 3]
```diff
@@ -133,4 +142,16 @@ class TicketsFragment : Fragment() {
...
+        builder.setMessage(resources.getString(R.string.no_tickets_message))
+               .setTitle(resources.getString(R.string.whoops))
+               .setPositiveButton(resources.getString(R.string.ok)) { dialog, _ -> dialog.cancel() }
+        val alert = builder.create()
+        alert.show()
+    }
+
+    private fun totalTickets(): Int {
+        return tickeIdAndQty.sumBy { it.second }
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: No logic in fragments

**Comment by dreadpool2** [comment]
@iamareebjamal PR Updated

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt [hunk 4]
```diff
@@ -90,7 +90,7 @@ class TicketsFragment : Fragment() {
         })
 
         rootView.register.setOnClickListener {
-            if (totalTickets() != 0) {
+            if (ticketsViewModel.totalTickets(tickeIdAndQty) != 0) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsFragment.kt]: logic in viewmodel

File: app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt [hunk 5]
```diff
@@ -51,8 +51,11 @@ class TicketsViewModel(private val ticketService: TicketService,private val even
                 }))
     }
 
-    fun totalTickets(tickeIdAndQty: List<Pair<Int, Int>>): Int {
-        return tickeIdAndQty.sumBy { it.second }
+    fun totalTicketsEmpty(tickeIdAndQty: List<Pair<Int, Int>>): Boolean {
+        if (tickeIdAndQty.sumBy { it.second } == 0) {
+            return true
+        }
+        return false
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt]: return tickeIdAndQty.sumBy { it.second } == 0

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/ticket/TicketsViewModel.kt]: Updated PR

**Pull request merged**